### PR TITLE
Clean up some small things in the float code

### PR DESCRIPTION
**Pull request by InvincibleRMC**
# Pull Request

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Related Tickets & Documents

- Related Issue #
- Closes #

### CODE REVIEW

**Review by benjaminwp18 — APPROVED** [review]

**Comment by InvincibleRMC** [comment]
@benjaminwp18 I assume you want to test this on device b4 I merge it in?

File: src/float/surface_transceiver/surface_transceiver.ino [hunk 1]
```diff
@@ -21,13 +21,12 @@ bool printRFStatus = true;
 // Singleton instance of the radio driver
 RH_RF95 rf95(RFM95_CS, RFM95_INT);
 
-const uint8_t LEGAL_COMMANDS_LEN = 5;
-const String LEGAL_COMMANDS[LEGAL_COMMANDS_LEN] = {"submerge", "suck", "pump", "stop", "return"};
+const String LEGAL_COMMANDS[] = {"submerge", "suck", "pump", "stop", "return"};
```
Comment (InvincibleRMC) [inline src/float/surface_transceiver/surface_transceiver.ino:24]: @benjaminwp18 Should this be a String enum maybe?

File: src/float/float_transceiver/float_transceiver.ino [hunk 2]
```diff
@@ -434,7 +434,7 @@ void initRadio() {
 
   if (!rf95.init()) {
     Serial.println("RFM95 radio init failed");
-    while (1) {}
+    while (1);
```
Comment (benjaminwp18) [inline src/float/float_transceiver/float_transceiver.ino:437]: Google doesn't like this
image

File: src/float/surface_transceiver/surface_transceiver.ino (hunk 1 above)
Comment (benjaminwp18) [reply src/float/surface_transceiver/surface_transceiver.ino:24]: That's something that the solution to the new float ticket would implement if necessary

File: src/float/surface_transceiver/surface_transceiver.ino [hunk 3]
```diff
@@ -69,13 +68,13 @@ void loop() {
     Serial.println("Getting serial command...");
     String command = Serial.readString();
 
-    if (command.startsWith("submerge")) {
+    if (command.equals("submerge")) {
```
Comment (benjaminwp18) [inline src/float/surface_transceiver/surface_transceiver.ino:71]: Behavior change

File: src/float/surface_transceiver/surface_transceiver.ino [hunk 4]
```diff
@@ -159,8 +158,8 @@ bool receivePacket() {
       byteBuffer[PKT_IDX_PROFILE_HALF]);
     for (int i = PKT_HEADER_LEN; i < len;) {
       // Wonky pointer casting to convert four bytes into a long (time)
-      serialPrintf("%l,", *reinterpret_cast<uint32_t*>(byteBuffer + i));
-      i += sizeof(uint32_t);
+      serialPrintf("%l,", *reinterpret_cast<unsigned long*>(byteBuffer + i));
+      i += sizeof(unsigned long);
```
Comment (benjaminwp18) [inline src/float/surface_transceiver/surface_transceiver.ino:162]: Times are specifically returned as `unsigned long`s in Arduino

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]
Overall I don't want to make a bunch of changes to this rn

I'm hoping Eric or a freshman will take over the big ticket on it and do a bigger rewrite

File: src/float/surface_transceiver/surface_transceiver.ino (hunk 4 above)
Comment (InvincibleRMC) [reply src/float/surface_transceiver/surface_transceiver.ino:162]: ?

File: src/float/surface_transceiver/surface_transceiver.ino (hunk 3 above)
Comment (InvincibleRMC) [reply src/float/surface_transceiver/surface_transceiver.ino:71]: True but, currently a `submerge foo bar` would be valid which seems like unecssary.

Comment (benjaminwp18) [inline src/float/surface_transceiver/surface_transceiver.ino:71]:
```suggestion
    if (command.startsWith("submerge")) {
```

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: src/float/surface_transceiver/surface_transceiver.ino [hunk 5]
```diff
@@ -69,13 +68,13 @@ void loop() {
...
-    if (command.startsWith("submerge")) {
+    if (command.equals("submerge")) {
       printRFStatus = false;
     }
 
-    for (int i = 0; i < LEGAL_COMMANDS_LEN; i++) {
-      if (command.startsWith(LEGAL_COMMANDS[i])) {
-        sendCommand(LEGAL_COMMANDS[i]);
+    for (auto legalCommand : LEGAL_COMMANDS) {
+      if (command.equals(legalCommand)) {
```
Comment (benjaminwp18) [inline src/float/surface_transceiver/surface_transceiver.ino:76]:
```suggestion
      if (command.startsWith(legalCommand)) {
```

**Pull request merged**